client/API.md

```diff
@@ -0,0 +1,536 @@
+    "enabled" : true,
+    "executing" : true,
+    "error" : "",
+    "stats": {}
+}
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 200  | Success             |
+| 404  | Task does not exist |
+
+
+### Enable Task
+
+To enable a task and start it executing make a POST request to the `/task/TASK_NAME/enable` endpoint.
+
+
+#### Example
+
+```
+POST /task/TASK_NAME/enable
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 204  | Success             |
+| 404  | Task does not exist |
+
+### Disable Task
+
+To disable a task and stop it executing make a POST request to the `/task/TASK_NAME/disable` endpoint.
+
+#### Example
+
+```
+POST /task/TASK_NAME/disable
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 204  | Success             |
+| 404  | Task does not exist |
+
+
+### Delete Task
+
+To delete a task make a DELETE request to the `/task/TASK_NAME` endpoint.
+
+```
+DELETE /task/TASK_NAME
+```
+
+#### Response
+
+| Code | Meaning             |
+| ---- | -------             |
+| 204  | Success             |
+
+>NOTE: Deleting a non-existent task is not an error and will return a 204 success.
+
+
+### List Tasks
+
+To get information about several tasks make a GET request to the `/task` endpoint.
+
+| Query Parameter | Purpose                                                                                                                                           |
+| --------------- | -------                                                                                                                                           |
+| pattern         | Filter results based on the pattern. Uses standard shell glob matching, see [this](https://golang.org/pkg/path/filepath/#Match) for more details. |
+
+#### Example
+
+Get all tasks.
+
+```
+GET /task
+```
+
+```
+{
+    "tasks" : [
+        {
+            "name" : "TASK_NAME",
+            "type" : "stream",
+            "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+            "enabled" : true,
+            "executing" : true,
+            "stats": {}
+        },
+        {
+            "name" : "ANOTHER_TASK_NAME",
+            "type" : "batch",
+            "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+            "enabled" : true,
+            "executing" : true,
+            "stats": {}
+        }
+    ]
+}
+```
+
+Optionally specify a glob `pattern` to list only matching tasks.
+
+```
+GET /task?pattern=TASK*
+```
+
+```
+{
+    "tasks" : [
+        {
+            "name" : "TASK_NAME",
+            "type" : "stream",
+            "dbrps": [{"db": "DATABASE_NAME", "rp" : "RP_NAME"}],
+            "enabled" : true,
+            "executing" : true,
+            "stats": {}
+        }
+    ]
+}
+```
+
+#### Response
+
+| Code | Meaning                |
+| ---- | -------                |
+| 200  | Success                |
+| 404  | No tasks match pattern |
```

Comment:
I think 404 should be for resources that truly don't exist.  I think this should return an empty list of tasks.

```json
{
    "tasks" : []
}
```

... this makes client code way easier... rather than if statements checking status codes you simply iterate over the response.